File: csharp/difference-of-squares/DifferenceOfSquares.cs
public static class DifferenceOfSquares
{
    public static int CalculateSquareOfSum(int max)
    {
        int sum = 0;
        for (int i = 1; i <= max; i++)
        {
            sum += i;
        }
        return (int)Math.Pow(sum, 2.0);
    }

    public static int CalculateSumOfSquares(int max)
    {
        int s = 0;
        for (int i = 1; i <= max; i++)
        {
            s += i * i;
        }
        return s;
    }

    public static int CalculateDifferenceOfSquares(int max)
    {
        int squareOfSum = CalculateSquareOfSum(max);
        int sumOfSquares = CalculateSumOfSquares(max);
        return squareOfSum >= sumOfSquares ? squareOfSum - sumOfSquares : sumOfSquares - squareOfSum;
    }
}

File: kotlin/difference-of-squares/src/main/kotlin/DifferenceOfSquares.kt
class Squares(val n: Int) {
    fun sumOfSquares(): Int = this.n * (this.n + 1) * (2 * this.n + 1) / 6

    fun squareOfSum(): Int {
        val sum = this.n * (this.n + 1) / 2
        return sum * sum
    }

    fun difference(): Int = squareOfSum() - sumOfSquares()
}
